### PR TITLE
Feat/spell check on content

### PR DESCRIPTION
**Pull request by galadd**
Description:
This pull request addresses https://github.com/MLH-Fellowship/orientation-project-python-23.SUM.A/issues/11 Spell check on content added by user. It aims to check that Experience, Education and Skill do not contain any spelling errors.

Key Changes:
The spell_check function has been written to utils.py, and function implementation to function experience, education and skill for both post and put method request in app.py.

Benefits:
This would help in situations where a user has made an error during inputs. The function returns its correction

Testing:
The test covers the format of return, which is to return the misspelled word alongside the sentence or phrase it was in. Test covers the post and put request for each of experience, education and skill

### CODE REVIEW

**Comment by wrussell1999** [comment]
@galadd before this gets merged, can you clarify how this works and what it's actually doing to check spelling?

**Comment by galadd** [comment]
Hello @wrussell1999 

The user input is checked for spelling errors before any other operations are performed on it. This is done for POST and PUT methods. (Right after `req = request.get_json()`). It iterates through each word, and check s if the spelling is in the dictionary. It ignores cases like logo links, special names and numbers. If there's a spelling error, it returns this format
```
[    
    {        
        "before": "Hellq there",        
        "after": "Hello there"    
    }
]

```

**Comment by Henoven** [comment]
> Hello @wrussell1999
> 
> The user input is checked for spelling errors before any other operations are performed on it. This is done for POST and PUT methods. (Right after `req = request.get_json()`). It iterates through each word, and check s if the spelling is in the dictionary. It ignores cases like logo links, special names and numbers. If there's a spelling error, it returns this format
> 
> ```
> [    
>     {        
>         "before": "Hellq there",        
>         "after": "Hello there"    
>     }
> ]
> ```
Firstly, congratulations on the PR @galadd ! It has a great description, behavior testing, and improved O(n) efficiency.

Regarding the behavior, I want to make sure I understand it correctly and address some potential edge cases. It seems that the feature performs a spelling check every time a user makes a POST or PUT request, which is fantastic. However, I noticed that we don't currently provide an option for the user to request this review. In issue #11, it was mentioned that the user should be able to request this revision. With the current implementation, users don't have the choice to opt out of a spelling review. What if they don't want a review because they are using acronyms or words in a language other than English? Are we considering these scenarios?

To tackle these concerns, I have a couple of ideas for simple solutions that can cover these edge cases:

- Create an endpoint specifically for requesting a spelling review. This way, users will have the opportunity to choose whether they want a review or not.
- Add a spelling flag to the existing endpoints, giving users the ability to decide if they want a review or not.

Let me know what you think about these suggestions!

**Comment by galadd** [comment]
> Let me know what you think about these suggestions!

This makes sense. My initial approach to it was to leave values that could potentially be acronyms or in other languages unchanged. But I'll look to the specifically requesting for a review

**Comment by galadd** [comment]
Hello @Henoven 

Thank you very much for the review
I have just added the update. Please review, and let me know if there's anything else I need to clarify

File: app.py [hunk 1]
```diff
@@ -90,6 +90,12 @@ def handle_post_experience():
 
     req = request.get_json()
 
+    choice = request.args.get("corrections")
```
Comment (Henoven) [inline app.py:93]: Hey @galadd 
What happens if the user doesn't send the corrections param?
The get method will return `None` by default if it can't find the key `corrections` you are looking for, so if the user doesn't sends the `corrections` param your condition might broke because you will be trying to compare `None = <string>` which will result in a `TypeError`  because `None` is a special object representing the absence of a value, while a string is a sequence of characters

```
none_value = None
string_value = "Hello"

# Comparing None with a string
result = none_value == string_value  # Raises a TypeError

```

File: app.py [hunk 2]
```diff
@@ -224,23 +250,51 @@ def skill():
...
-        code, err_message = validate_request(req, required_fields)
+    return jsonify({"Server Error": "Couldn't process method"})
 
-        if code != 0:
-            return jsonify({"error": err_message}), code
+def handle_get_skill():
+    '''
+    Handles skill get request
+    '''
+    index = request.args.get("index")
```
Comment (Henoven) [inline app.py:273]: I love that you modularized the code, well done!

There is something I would like to suggest, pass the params directly to the functions `handle_get_skill(request.args.get('index'))` to avoid conflicts

File: test_pytest.py [hunk 3]
```diff
@@ -309,3 +309,190 @@ def test_edu_update():
                                   json=updated_example_education)
     response = app.test_client().get('/resume/education')
     assert response.json[item_id] == updated_example_education
+
```
Comment (Henoven) [inline test_pytest.py:312]: Well done!
Great tests!

It's recommendable to test edge cases. Please create a test for when the user doesn't send the corrections parameter.

**Review by Henoven — CHANGES_REQUESTED** [review]
Hey @galadd, this PR is looking great!

I really appreciate your implementation and how you approached this solution. With these corrections, I believe we're almost there. I want to emphasize that my intention is not to undermine the work you've done. The comments I provided are meant to help you think beyond the issue description. In real-life scenarios, issues can be reported by users who may not fully understand how the project works. That's why it's important to communicate with the maintainer and your team to consider all the edge cases, in order to have a clear understanding of the objective. I'm certain you will do fantastic.

Keep up the good work!

File: app.py (hunk 1 above)
Comment (galadd) [reply app.py:93]: If corrections parameter is not passed. The function continues without a return corrections

File: test_pytest.py (hunk 3 above)
Comment (galadd) [reply test_pytest.py:312]: The last test covers for no corrections parameter passed. Check out `test_spell_check_no_corrections()`

File: app.py (hunk 2 above)
Comment (galadd) [reply app.py:273]: I wanted to maintain the consistency of other fellows' code. There's also a check `if index is not None`. The alternative will be to write something like `if (request.args.get('index')) is not None`. I think the first style is more readable and also maintains the code consistency, while the second style is more ambiguous. Please, let me know what you think about this

**Comment by galadd** [comment]
@Henoven
